src/LargeFileUploadTask.ts

```diff
@@ -0,0 +1,338 @@
+/**
+ * @module LargeFileUploadTask
+ */
+
+import { Client } from "./index";
+import { Range } from "./Range";
+import { getValidRangeSize } from "./LargeFileUploadUtil";
+
+/**
+ * User Options to create the upload task
+ */
```

Comment:
Consider this comment applicable to all of the files in this PR. Let's add doc comments to all of these types and members. These are all intended to be used by customers so let's make it easier for them to use them. Our .d.ts should be picking these up as well. I'm not sure how we go from .ts to .d.ts. Is there a tool for that? 

/**
 * Interface used to define options when creating an upload task.
 * @interface
 * @property {string} sessionRequestUrl - The default values for parties.
 * @property {string}  fileName               - Optional. Specifies the file name of the file to upload. A file name will be created for you if you do not specify a value.
 * @property {number}  rangeSize          - Optional. Specifies the range chunk size. The default value of ??? is used if a value is not specified. 
 * @property {number} maxTries             - Optional. Specifies the ??? The default value of ??? is used if a value is not specified.
 */

Reply:
sure, will define interface in this format.

Typescript compiling does that, when we do compile .ts files using typescript compiler, it generates .js, .d.ts and .js.map files

will investigate and make the compiler take comments from .ts to .d.ts if possible.